cmd/ghalistener/metrics/metrics.go

```diff
@@ -152,13 +152,145 @@ type ExporterConfig struct {
+	}
+	if e.Metrics != nil {
+		return
+	}
+
+	e.Metrics = &v1alpha1.MetricsConfig{
+		Counters: map[string]*v1alpha1.CounterMetric{
+			MetricStartedJobsTotal: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyJobName,
+					labelKeyEventName,
+				},
+			},
+			MetricCompletedJobsTotal: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyJobName,
+					labelKeyEventName,
+					labelKeyJobResult,
+				},
+			},
+		},
+		Gauges: map[string]*v1alpha1.GaugeMetric{
+			MetricAssignedJobs: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyRunnerScaleSetName,
+					labelKeyRunnerScaleSetNamespace,
+				},
+			},
+			MetricRunningJobs: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyRunnerScaleSetName,
+					labelKeyRunnerScaleSetNamespace,
+				},
+			},
+			MetricRegisteredRunners: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyRunnerScaleSetName,
+					labelKeyRunnerScaleSetNamespace,
+				},
+			},
+			MetricBusyRunners: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyRunnerScaleSetName,
+					labelKeyRunnerScaleSetNamespace,
+				},
+			},
+			MetricMinRunners: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyRunnerScaleSetName,
+					labelKeyRunnerScaleSetNamespace,
+				},
+			},
+			MetricMaxRunners: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyRunnerScaleSetName,
+					labelKeyRunnerScaleSetNamespace,
+				},
+			},
+			MetricDesiredRunners: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyRunnerScaleSetName,
+					labelKeyRunnerScaleSetNamespace,
+				},
+			},
+			MetricIdleRunners: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyRunnerScaleSetName,
+					labelKeyRunnerScaleSetNamespace,
+				},
+			},
+		},
+		Histograms: map[string]*v1alpha1.HistogramMetric{
+			MetricJobStartupDurationSeconds: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyJobName,
+					labelKeyEventName,
+				},
+				Buckets: defaultRuntimeBuckets,
+			},
+			MetricJobExecutionDurationSeconds: {
+				Labels: []string{
+					labelKeyEnterprise,
+					labelKeyOrganization,
+					labelKeyRepository,
+					labelKeyJobName,
+					labelKeyEventName,
+					labelKeyJobResult,
+				},
+				Buckets: defaultRuntimeBuckets,
+			},
+		},
+	}
 }
 
 func NewExporter(config ExporterConfig) ServerExporter {
+	config.defaults()
```

Comment:
Do we want to add some more tests in `metrics_test.go` to cover all the possible configuration options?

Reply:
Added few tests :+1: